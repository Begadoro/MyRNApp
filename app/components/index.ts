export * from "./Screen"
export * from "./Button"
export * from "./CustomInput"
export * from "./Heading"
export * from "./CustomHeader"
export * from "./Home/CategorySquare"
export * from "./Home/ItemSquare"
export * from "./Badge"
export * from "./RoundIconButton"
export * from "./Loader"
