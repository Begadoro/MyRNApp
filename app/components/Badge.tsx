import * as React from "react"
import { StyleProp, StyleSheet, TouchableOpacity, ViewStyle, Text } from "react-native"
import { observer } from "mobx-react-lite"
import { colors, spacing } from "app/theme"
import { Feather } from "@expo/vector-icons"

export interface BafgeProps {
  /**
   * An optional style override useful for padding & margin.
   */
  style?: StyleProp<ViewStyle>
  bgColor?: string
  color?: string
  icon?: typeof Feather[keyof typeof Feather]
  text: string
  onPress?: () => void,
  hasDelete?: boolean
}

/**
 * Describe your component here
 */
export const Badge = observer(function Badge(props: BafgeProps) {

  const $style = getStyle(props.bgColor, props.color);

  return (
    <TouchableOpacity  style={[$style.container, props.style]} onPress={props.onPress}>
      {props.icon ?
        <Feather name={props.icon} size={spacing.sm} color={props.color ?? colors.text}/> : null
      }
      <Text style={$style.text}>{props.text}</Text>
      { props.hasDelete ? <Feather name={"x"} size={spacing.sm} color={props.color ?? colors.text}/> : null}
    </TouchableOpacity>
  )
})

const getStyle = (bgColor?: string, color?: string) => StyleSheet.create({
  container:{
    alignItems: "center",
    backgroundColor: bgColor ?? colors.errorBackground,
    borderRadius: spacing.sm,
    display: "flex",
    flexDirection: "row",
    gap: spacing.xxs,
    justifyContent: "center",
    paddingHorizontal: spacing.xs,
    paddingVertical: spacing.xxs,
    width: 80
  },
  text:{
    color: color ?? colors.text,
    fontSize: spacing.sm
  },
})