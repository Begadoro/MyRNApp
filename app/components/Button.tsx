import * as React from "react"
import { StyleProp, StyleSheet, TextStyle, TouchableOpacity, ViewStyle, Text } from "react-native"
import { observer } from "mobx-react-lite"
import { colors, spacing } from "app/theme"

export interface ButtonProps {
  /**
   * An optional style override useful for padding & margin.
   */
  style?: StyleProp<ViewStyle>
  text: string,
  textStyle?: StyleProp<TextStyle>
  onPress?: () => void
}

/**
 * Standard customizable button. It takes 100% of width of his parent
 */
export const Button = observer(function Button(props: ButtonProps) {

  const style = getStyles();

  return (
    <TouchableOpacity style={[style.container,props.style]} onPress={props.onPress}>
      <Text style={[style.text, props.textStyle]}>{props.text}</Text>
    </TouchableOpacity>
  )
})

const getStyles = () => StyleSheet.create({
  container:{
    alignItems: "center",
    backgroundColor: colors.buttonsBg,
    borderRadius: spacing.lg,
    height: 44,
    justifyContent: "center",
    width: "100%"
  },
  text:{
    color: colors.text,
    fontSize: spacing.md,
  }
})