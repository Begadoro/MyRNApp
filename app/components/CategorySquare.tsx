import * as React from "react"
import { StyleProp, StyleSheet, Image, Text, ViewStyle, TouchableOpacity } from "react-native"
import { observer } from "mobx-react-lite"
import { spacing } from "app/theme"


export interface CategorySquareProps {
  /**
   * An optional style override useful for padding & margin.
   */
  style?: StyleProp<ViewStyle>
  onPress?: () => void,
  image: any,
  text: string
}

/**
 * Describe your component here
 */
export const CategorySquare = observer(function CategorySquare(props: CategorySquareProps) {

  const style = getStyles();

  return (
    <TouchableOpacity onPress={props.onPress} style={[style.container,props.style]}>
      <Image source={props.image} style={style.image}/>
      <Text style={style.text}>{props.text}</Text>
    </TouchableOpacity>
  )
})

const getStyles = () => StyleSheet.create({
  container:{
    alignItems: "center",
    display: "flex",
    flexDirection: "column",
    gap: spacing.xs,
    justifyContent: "center",
  },
  image:{
    borderRadius: spacing.md,
    height: 110,
    width: 110,
  },
  text:{
    fontSize: spacing.sm
  }
})
