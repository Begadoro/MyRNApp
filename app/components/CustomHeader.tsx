import { Platform, StyleProp, StyleSheet, ViewStyle } from "react-native"
import { observer } from "mobx-react-lite"
import Animated from "react-native-reanimated";
import React, {ReactNode} from "react";
import Constants from "expo-constants";
import { colors, spacing } from "app/theme"


export interface CustomHeaderProps {
  /**
   * An optional style override useful for padding & margin.
   */
  style?: StyleProp<ViewStyle>,
  animatedStyle?: StyleProp<ViewStyle>,
  animatedBackgroundStyle?: StyleProp<ViewStyle>,
  leftComponent?: ReactNode,
  rightComponent?: ReactNode,
}

/**
 * Custom header with left and right components.
 * Props are used to animate the header
 */
export const CustomHeader = observer(function CustomHeader(props: CustomHeaderProps) {

  const STATUS_BAR_HEIGHT = Constants.statusBarHeight;
  const PLATFORM = Platform.OS;
  const HEADER_HEIGHT = STATUS_BAR_HEIGHT + (PLATFORM === 'ios' ? 44 : 50);
  const PD_TOP = STATUS_BAR_HEIGHT - (PLATFORM === 'ios' ? 10 : 5);


  const $style = getStyle(HEADER_HEIGHT, PD_TOP);

  return(
    <Animated.View style={[$style.container, props.animatedStyle, props.animatedBackgroundStyle]}>
      {props.leftComponent ?? null}
      {props.rightComponent ?? null}
    </Animated.View>
  )
})

const getStyle = (HEADER_HEIGHT: number, PD_TOP: number) => StyleSheet.create({
  container:{
    alignItems: 'center',
    backgroundColor: colors.background,
    display: 'flex',
    flexDirection: 'row',
    height: HEADER_HEIGHT,
    justifyContent: 'space-between',
    paddingHorizontal: spacing.md,
    paddingTop: PD_TOP,
    width: '100%'
  }
})