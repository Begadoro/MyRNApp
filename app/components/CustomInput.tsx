import * as React from "react"
import { StyleProp, StyleSheet, TextInput, View, ViewStyle } from "react-native"
import { observer } from "mobx-react-lite"
import { colors, spacing } from "app/theme"
import { Feather } from "@expo/vector-icons"

export interface CustomInputProps {
  /**
   * An optional style override useful for padding & margin.
   */
  style?: StyleProp<ViewStyle>
  onBlur?: () => void
  onChange: (value: string) => void
  value: string
  placeholder?: string
  type?: "password" | "text",
  errors?: any
  icon? : typeof Feather[keyof typeof Feather]
  onEndEditing?: () => void
}

/**
 * Describe your component here
 */
export const CustomInput = observer(function CustomInput(props: CustomInputProps) {

  const style = getStyles();

  return (
    <View style={[style.root, props.errors && style.error]}>
      {props.icon ? <Feather name={props.icon} size={spacing.md} color={props.errors ? "red" : colors.border}/> : null}
      <TextInput
        style={style.container}
        onBlur={props.onBlur}
        onChangeText={(value) => props.onChange(value)}
        value={props.value}
        placeholder={props.placeholder ?? "Enter text"}
        placeholderTextColor={colors.border}
        secureTextEntry={props.type === "password"}
        onEndEditing={props.onEndEditing}
      />
    </View>

  )
})

const getStyles = () => StyleSheet.create({
  container:{
    color: colors.text,
    height: "100%",
    width: "90%"
  },
  error:{
    borderColor: "red",
    borderWidth: 1,
    color: "red",
  },
  root:{
    alignItems: "center",
    backgroundColor:"#dfdfdf",
    borderRadius: spacing.md,
    display: "flex",
    flexDirection: "row",
    gap: spacing.sm,
    height: 44,
    paddingHorizontal: spacing.md,
    width: "100%",
  },
})