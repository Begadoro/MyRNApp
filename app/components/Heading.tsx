import * as React from "react"
import { StyleProp, View, ViewStyle, StyleSheet, Text } from "react-native"
import { observer } from "mobx-react-lite"
import { colors, spacing } from "app/theme"
import { bold, h1, h2, h3 } from "app/utils/constants"
import { Feather } from "@expo/vector-icons"

export interface HeadingProps {
  /**
   * An optional style override useful for padding & margin.
   */
  style?: StyleProp<ViewStyle>,
  text: string
  fontColor?: string
  iconColor?: string
  icon?: typeof Feather[keyof typeof Feather]
  h?: string
}

/**
 * Describe your component here
 */
export const Heading = observer(function Heading(props: HeadingProps) {

  const $style = getStyle(props.fontColor ?? colors.text);

  const headings = new Map([
    ['h1',h1],
    ['h2',h2],
    ['h3',h3],
  ])
  const heading = props.h ? headings.get(props.h) : h1;

  return (
    <View style={[$style.container, props.style]}>
      {props.icon ? <Feather  name={props.icon} size={spacing.lg} color={props.iconColor ?? colors.text}/> : null}
      <Text style={[heading, bold, $style.text]}>{props.text}</Text>
    </View>
  )
})

const getStyle = (fontColor: string) => StyleSheet.create({
  container:{
    alignItems: "center",
    display: "flex",
    flexDirection: "row",
    gap: spacing.xxs
  },
  text:{
    color: fontColor
  }
})