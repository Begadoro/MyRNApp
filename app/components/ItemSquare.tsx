import * as React from "react"
import { Image, Text, StyleProp, StyleSheet, TouchableOpacity, View, ViewStyle } from "react-native"
import { observer } from "mobx-react-lite"
import { spacing } from "app/theme"

export interface ItemSquareProps {
  /**
   * An optional style override useful for padding & margin.
   */
  style?: StyleProp<ViewStyle>
  onPress?: () => void,
  image: any,
  text: string
  price: number
}

/**
 * Describe your component here
 */
export const ItemSquare = observer(function ItemSquare(props: ItemSquareProps) {

  const style = getStyles();

  return (
    <TouchableOpacity onPress={props.onPress} style={[style.container,props.style]}>
      <Image source={props.image} style={style.image}/>
      <View>
        <Text style={style.text}>{props.text}</Text>
        <Text style={[style.text, style.textBold]}>{props.price}â‚¬</Text>
      </View>
    </TouchableOpacity>
  )
})

const getStyles = () => StyleSheet.create({
  container:{
    alignItems: "flex-start",
    display: "flex",
    flexDirection: "column",
    gap: spacing.xs,
    justifyContent: "center",
  },
  image:{
    borderRadius: spacing.md,
    height: 150,
    width: "100%",
  },
  text:{
    fontSize: spacing.md
  },
  textBold:{
    fontWeight: "bold"
  }
})
