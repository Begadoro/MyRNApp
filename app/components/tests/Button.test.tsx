import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from "app/components"

test('renders correctly', () => {
    const tree = renderer.create(<Button text="test"/>).toJSON();
    expect(tree).toMatchSnapshot();
})