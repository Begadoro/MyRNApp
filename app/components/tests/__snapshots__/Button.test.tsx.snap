// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#621893",
      "borderRadius": 24,
      "height": 44,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#191015",
          "fontSize": 16,
        },
        undefined,
      ]
    }
  >
    test
  </Text>
</View>
`;
